docs/dev-guide/design/extension-architecture.md

```
@@ -0,0 +1,195 @@
+**Usage:**
+```bash
+lola ext add ./hello-target/
+lola install my-module -a hello-target
+```
+
+## Hello World: Python Repo Extension
+
+A repo extension providing search and resolve for a custom skill catalog.
+
+**Directory structure:**
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)